src/engine.ts

```diff
@@ -788,12 +820,31 @@ export default class Engine {
     this.endTurnPhase(player, Command.Build);
   }
 
+
+  [Command.Special](player: PlayerEnum, income: string){
+    const specialEvent = Event.parse([income]);
+    const { specialacts } = this.availableCommand(player, Command.Special).data;
+    const actAvailable = specialacts.find(sa => sa.income == income);
+    
+    assert(actAvailable !== undefined, `Special action ${income} is not available`);
+
+    //gets income for standard once per turn activated income
+    this.player(player).gainRewards(specialEvent[0].rewards);
+    //mark as activated special action for this turn
+    this.player(player).activateEvents(specialEvent, true);
+
+    if ( income === "=>range" || income === "=>step" ){
```

Comment:
This shouldn't be needed, It's already handled in `Player.gainRewards`

Reply:
I changed it. We don't call `buildMinePhase`  but still need to do not call `endTurnPhase`